core/modules/qana/QservRestrictorPlugin.cc

```diff
@@ -64,27 +64,354 @@
+                " (must be even), will not apply an area restrictor..");
+        return nullptr;
+    }
+    return std::make_shared<query::QsRestrictor>(name, parameters);
+}
+
+
+std::shared_ptr<query::QsRestrictor> makeQsRestrictor(query::FuncExpr const& scisqlFunc) {
+    std::vector<std::string> parameters;
+    int counter(0);
+    for (auto const& valueExpr : scisqlFunc.getParams()) {
+        if (counter++ < 2) {
+            // The first 2 parameters are the ra and decl columns to test; these get thrown away.
+            continue;
+        }
+        if (valueExpr->isConstVal()) {
+            parameters.push_back(valueExpr->getConstVal());
+        } else {
+            // If any parameter in the scisql restrictor function is not a const value then we can't use it
+            // (for example, we don't support math functions in the area restrictor.)
+            // Give up & carry on.
+            return nullptr;
+        }
+    }
+    if (scisqlFunc.getName() == "scisql_s2PtInBox") {
+        return makeQsRestrictor("qserv_areaspec_box", parameters, 4);
+    } else if (scisqlFunc.getName() == "scisql_s2PtInCircle") {
+        return makeQsRestrictor("qserv_areaspec_circle", parameters, 3);
+    } else if (scisqlFunc.getName() == "scisql_s2PtInEllipse") {
+        return makeQsRestrictor("qserv_areaspec_ellipse", parameters, 5);
+    } else if (scisqlFunc.getName() == "scisql_s2PtInCPoly") {
+        return makeQsRestrictor("qserv_areaspec_poly", parameters, 6, true, true);
+    }
+    return nullptr;
+}
+
+
+/**
+ * @brief Add scisql restrictors for each QsRestrictor.
+ *
+ * This handles the case where a qserv areaspec function was passed into the WHERE clause by the user,
+ * it adds scisql restrictor functions corresponding to the qserv area restrictor that is applied as a
+ * result of the areaspec function.
+ *
+ * @param restrictors The qserv area restrictors to add scisql area restrictors for.
+ * @param whereClause The query's WHERE clause, to which scisql area restrictors will be added.
+ * @param fromList The query's FROM list, it is consulted to find the chunked tables.
+ * @param context The query context.
+ */
+void addScisqlRestrictors(std::vector<std::shared_ptr<query::QsRestrictor>> const& restrictors,
+                           query::FromList const& fromList,
+                           query::WhereClause& whereClause,
+                           query::QueryContext& context) {
+    if (restrictors.empty()) return;
+
+    auto const& tableList = fromList.getTableRefList();
+    RestrictoryEntryList chunkedTables;
+    getTable gt(*context.css, chunkedTables);
+    std::for_each(tableList.begin(), tableList.end(), gt);
+    // chunkedTables is now populated with a RestrictorEntry for each table in the FROM list that is chunked.
+    if (chunkedTables.empty()) {
+        throw qana::AnalysisError("Spatial restrictor without partitioned table.");
```

Comment:
I thought that any spatial restrictors will be simply ignored for non-partitioned (non-chunked) tables as per its, `2` of the proposal in:
https://jira.lsstcorp.org/browse/DM-17295

Reply:
My takeaway was if there is a `qserv_areaspec_` restrictor, then use 'old' behavior (including throwing if there is not a chunked table. Otherwise, do the new thing.

Reply:
(this code handles the 'old' case)